start_simulation.sh

```diff
@@ -3,4 +3,67 @@
 SCRIPT=$(realpath "$0")
 SCRIPTDIR=$(dirname "${SCRIPT}")
 
-(cd ${SCRIPTDIR} && ${SCRIPTDIR}/tools/bazelisk run //spines:bullet_spine -- --show)
+URL_ARCHIVE="https://github.com/pgraverdy/upkie/releases/download"
+
+# no v at start of version number here
+VERSION=$(awk '/^PROJECT_NUMBER/{print $3}' docs/Doxyfile)
+
+SYSTEM=$(uname -s)
+ARCH=$(uname -m)
+
+if [[ "$SYSTEM" == Darwin ]]; then
+    echo "MacOS system"
+    if [[ "$ARCH" == x86_64* ]]; then
+        echo "X64 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_x86_bullet_spine.tar.gz
+        DIR_ARCHIVE="darwin_x86_bullet_spine"
+    elif [[ "$ARCH" == i*86 ]]; then
+        echo "X32 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_x86_bullet_spine.tar.gz
+        DIR_ARCHIVE="darwin_x86_bullet_spine"
+    elif  [[ "$ARCH" == arm* ]]; then
+        echo "ARM Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/darwin_arm64_bullet_spine.tar.gz
+        DIR_ARCHIVE="darwin_arm64_bullet_spine"
+    else
+        echo "Unsupported architecture $ARCH"
+    fi
+elif  [[ "$SYSTEM" == Linux ]]; then
+    echo "Linux system"
+    if [[ "$ARCH" == x86_64* ]]; then
+        echo "X64 Architecture"
+        SPINE_ARCHIVE="$URL_ARCHIVE"/v"$VERSION"/linux_amd64_bullet_spine.tar.gz
+        DIR_ARCHIVE="linux_amd64_bullet_spine"
+    else
+        echo "Unsupported architecture $ARCH"
+    fi
+else
+    echo "Unsupported System $SYSTEM"
+fi
+
+if [[ -z "$SPINE_ARCHIVE" ]]; then
+    echo "Build spine and use";
+    (cd ${SCRIPTDIR} && ${SCRIPTDIR}/tools/bazelisk run //spines:bullet_spine -- --show)
+else
+    echo "Using SPINE_ARCHIVE:$SPINE_ARCHIVE"
+    mkdir tmp-bin
```

Comment:
Another pro of `mktemp` is that we don't need to check whether the directory exists (otherwise here we should do a `mkdir -p` in case the user re-runs the script).

Reply:
Should we then do a pushd/popd (and also print the temp dir in case the user wants to go back ?)

Reply:
Just doing mktemp (and cd in it)